mesonbuild/scripts/python_info.py

```diff
@@ -63,10 +62,15 @@ def get_install_paths():
     return paths, install_paths
 
 def links_against_libpython():
-    from distutils.core import Distribution, Extension
-    cmd = Distribution().get_command_obj('build_ext')
-    cmd.ensure_finalized()
-    return bool(cmd.get_libraries(Extension('dummy', [])))
+    # on versions supporting python-embed.pc, this is the non-embed lib
+    if sys.version_info >= (3, 12):
+        variables = sysconfig.get_config_vars()
+        return bool(variables.get('LIBPYTHON', True))
```

Comment:
I've changed this to this form from
```python
return bool(variables.get('LIBPYTHON', 'yes'))
```
because the `yes` there always makes me pause and wonder if someone expects `LIBPYTHON` to be set to `no` and have `bool()` return false in that case. It is clearly not the case, but `True` avoids the mental overhead.